modules/react/color-picker/lib/ColorInput.tsx

```diff
@@ -109,29 +116,52 @@ const SwatchTile = styled(ColorSwatch)({
   pointerEvents: 'none',
 });
 
-export default class ColorInput extends React.Component<ColorInputProps> {
-  public render() {
-    // TODO: Standardize on prop spread location (see #150)
-    const {
+const isValidHex = (value: string) => {
+  return /(^#?[0-9A-F]{3}$)|(^#?[0-9A-F]{6}$)/i.test(value);
+};
+
+const formatValue = (value: string) => {
+  return value.replace(/#/g, '').substring(0, 6);
+};
+
+export const ColorInput = createComponent('input')({
+  displayName: 'ColorInput',
+  Component: (
+    {
       placeholder = 'FFFFFF',
       value = '',
       showCheck,
       onChange,
       onValidColorChange,
-      inputRef,
       disabled,
       error,
       grow,
       ...elemProps
-    } = this.props;
-    const formattedValue = this.formatValue(value);
+    }: ColorInputProps,
+    ref,
+    Element
+  ) => {
+    const formattedValue = formatValue(value);
+
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+      const value = formatValue(e.currentTarget.value);
+
+      if (onChange) {
+        onChange(e);
+      }
```

Comment:
Comment (not an RC) I _think_ `createComponent` will handle merging this `onChange` for you. I'd need to poke at the code a bit to be sure though. Maybe something to look at. Your call.

Reply:
`createComponent` doesn't take care of merging props, `mergeProps` does. Something like:

```tsx
const inputProps = mergeProps({
 onChange: handleChange,
  // other props you want
}, elemProps)
```

Then on `<CustomHexInput>`, you'd do something like:
```tsx
<ColorHexInput
  dir="ltr"
  // ...
  {...inputProps}
>
```

I'm not sure I'd worry about it though. I think it would be better to turn this into a compound component with behavior hooks that are exported which handle prop merging better.

If you wanted, this line could be simplified to:

```tsx
onChange?.(e)
```

Reply:
Oh I was under the impression it was merging the css prop and other event handler functions. I’ll look back at the source again.